### DISCUSSION TITLE
Add 'MediaProfilePicker' to TextField editor options?

### DISCUSSION THREAD

**Discussion by netwavebe** [OrchardCMS/OrchardCore#7466] [Ideas]
I think media profiles are a good thing because they can be used by non-technical users.

I would like to show the available media profiles in a dropdown when adding/editing certain content items for a project I'm working on. I was thinking of adding an editor option 'Media Profile picker' to the TextField to achieve this.

It would look like this:
![image](https://user-images.githubusercontent.com/3008547/97802302-c268cf00-1c42-11eb-8b95-73d294add4f4.png)

Is this something I should create in `OrchardCore.Media` so everyone can use this? Or rather put it in my own module?

I think @deanmarcussen is the author of media profiles? How do you see this, Dean?

**Comment by deanmarcussen** [comment]
We talked about this during one of the recent demos of the profiles / media text feature.

The general feeling was that a profile was more of a template provided value, rather than a ui value.

For example, imagine if you have these profiles
- banner
- small
- medium

Then you have a picker as you suggest, but the page you are picking for only has space for small and medium.
And the user picks a banner, and wonders why it doesn't fit properly (or turn into a banner)

So that was why we didn't want to do one.

Now I realise that the scenario above is not always true.

BUT, why don't you just use a `PredefinedList` editor? 

Then you are able to set the profiles that image can support? 
... and it has nice key value pair support

**Comment by netwavebe** [comment]
All valid arguments, I can see the reasoning behind this.

My case is slightly more generic as it's shared on multiple websites (with other profiles) but even then not all profiles are always valid choices. A predefinedList, as you suggest, might be better.
